src/main/java/com/amazon/opendistroforelasticsearch/sql/rewriter/matchtoterm/TermFieldRewriter.java

```diff
@@ -210,19 +214,53 @@ public boolean isValidIdentifierForTerm(SQLIdentifierExpr expr) {
             );
     }
 
-    public boolean hasIdenticalMappings(TermFieldScope scope, Map<String, String> indexToType) {
+    public void checkMappingCompatibility(TermFieldScope scope, Map<String, String> indexToType) {
```

Comment:
can this method be `private`?

Reply:
Yes, it can. But I saw many methods that has only one usage within the same class are given public access identifier(even the one I deleted(hasIdenticalMappings), I wonder if you did this on purpose for future use.

Reply:
`public` methods are meant to be called from outside, if there is no need for that ,the method should not be `public`. In this case `private` is the best option, since this is just internal implementation.

Reply:
That makes sense. Fixed.